model/godleyTable.h

```diff
@@ -72,21 +70,23 @@ namespace minsky
       : m_assetClass(other.m_assetClass),
         data(other.data),
         doubleEntryCompliant(other.doubleEntryCompliant),
-        multipleEquities(other.multipleEquities),
         title(other.title)
     { }
   
     // Perform deep comparison of Godley tables in history to avoid spurious noAssetClass columns from arising during undo. For ticket 1118.
     bool operator==(const GodleyTable& other) const 
     {
 		return (data==other.data && m_assetClass==other.m_assetClass &&
-		 doubleEntryCompliant==other.doubleEntryCompliant && multipleEquities==other.multipleEquities && title==other.title);
+		 doubleEntryCompliant==other.doubleEntryCompliant && title==other.title);
     }    
 
     /// class of each column (used in DE compliant mode)
     const vector<AssetClass>& _assetClass() const {return m_assetClass;}
     AssetClass _assetClass(size_t col) const;
     AssetClass _assetClass(size_t col, AssetClass cls);
+    
+    // Check whether more than one equity column is present irrespective of single or multiple equity column mode. For feature 111.
+    bool singleEquity();
```

Comment:
This method should be const

Reply:
Will I have to add a TCL accessor getter as in the case of assetClass in godleyTable.h?

Reply:
It looks like this will require a class_desc modification, so not sure about doing so. I tried:

    bool m_multipleEquities=false;

public:
    // Allow multiple equity columns.
    const bool& _multipleEquities() const {return m_multipleEquities;}
    bool _multipleEquities(const bool& me)
    {
		m_multipleEquities=me;
		return m_multipleEquities;
	}
    
    /**
       TCL accessor method 
       @param me - boolean value of multipleEquities member
       @return current value of multipleEquities
       sets if value is passed, otherwise gets
    */
    bool multipleEquities(ecolab::TCL_args args); 

But got an undefined reference at the linking stage:

minskyTCL.o: In function `void classdesc_access::access_isa<minsky::Minsky>::operator()<minsky::Minsky>(std::set<std::type_info const*, std::less<std::type_info const*>, std::allocator<std::type_info const*> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, minsky::Minsky&)':
/home/wd/minsky/./minsky.cd:1059: undefined reference to `minsky::Minsky::multipleEquities(ecolab::TCL_args)'
minskyTCL.o: In function `void classdesc_access::access_TCL_obj<minsky::Minsky>::operator()<minsky::Minsky>(classdesc::TCL_obj_t&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, minsky::Minsky&)':
/home/wd/minsky/./minsky.cd:784: undefined reference to `minsky::Minsky::multipleEquities(ecolab::TCL_args)'
minskyTCL.o: In function `void classdesc_access::access_pack<minsky::Minsky>::operator()<minsky::Minsky>(classdesc::pack_t&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, minsky::Minsky&)':
/home/wd/minsky/./minsky.cd:186: undefined reference to `minsky::Minsky::multipleEquities(ecolab::TCL_args)'
minskyTCL.o: In function `void classdesc_access::access_unpack<minsky::Minsky>::operator()<minsky::Minsky>(classdesc::pack_t&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, minsky::Minsky&)':
/home/wd/minsky/./minsky.cd:435: undefined reference to `minsky::Minsky::multipleEquities(ecolab::TCL_args)'
collect2: error: ld returned 1 exit status
Makefile:162: recipe for target 'gui-tk/minsky' failed
make: *** [gui-tk/minsky] Error 1

So, I'm going back to non const.

Reply:
Could not achieve compilation with a const method. I think my lack of knowledge of and/or experience in C++ is the cause here...